src/mq.c

```diff
@@ -434,120 +439,76 @@ static int mq_publish_message(const char *exchange,
  * sent to your reply.
  *
  * Returns: 0 if successful and negative integer otherwise.
  */
-int8_t mq_publish_direct_message_rpc(const char *exchange,
-				     const char *routing_key,
-				     amqp_table_entry_t *headers,
-				     size_t num_headers,
-				     uint64_t expiration_ms,
-				     amqp_bytes_t reply_to,
-				     const char *correlation_id,
-				     const char *body)
-{
-	return mq_publish_message(exchange, AMQP_EXCHANGE_TYPE_DIRECT,
-				  routing_key, headers,
-				  num_headers, expiration_ms,
-				  reply_to, correlation_id, body);
-}
-
-/**
- * mq_publish_direct_message:
- * @exchange: exchange name
- * @routing_key: routing key name
- * @headers: array of table entry with headers
- * @num_headers: headers length
- * @expiration_ms: expiration property in miliseconds or 0 if no expiration time
- * @body: the message to be sent
- *
- * Publish a message with exchange type Direct
- *
- * Returns: 0 if successful and negative integer otherwise.
- */
-int8_t mq_publish_direct_message(const char *exchange,
-				 const char *routing_key,
-				 amqp_table_entry_t *headers,
-				 size_t num_headers,
-				 uint64_t expiration_ms,
-				 const char *body)
-{
-	return mq_publish_message(exchange, AMQP_EXCHANGE_TYPE_DIRECT,
-				  routing_key, headers,
-				  num_headers, expiration_ms,
-				  amqp_empty_bytes, NULL, body);
-}
-
-/**
- * mq_publish_fanout_message:
- * @exchange: exchange name
- * @headers: array of table entry with headers
- * @num_headers: headers length
- * @expiration_ms: expiration property in miliseconds or 0 if no expiration time
- * @body: the message to be sent
- *
- * Publish a message with exchange type Fanout.
- *
- * Returns: 0 if successful and negative integer otherwise.
- */
-int8_t mq_publish_fanout_message(const char *exchange,
-				 amqp_table_entry_t *headers,
-				 size_t num_headers,
-				 uint64_t expiration_ms,
-				 const char *body)
-{
-	return mq_publish_message(exchange, AMQP_EXCHANGE_TYPE_FANOUT,
-				  NULL, headers,
-				  num_headers, expiration_ms,
-				  amqp_empty_bytes, NULL, body);
+int8_t mq_publish_message(const mq_message_data_t *message) {
+	int8_t res;
+	switch (message->msg_type) {
+		case MQ_MESSAGE_TYPE_DIRECT:
+			res = mq_publish(message->exchange, AMQP_EXCHANGE_TYPE_DIRECT,
+					 message->routing_key, headers, num_of_headers,
+					 message->expiration_ms, amqp_empty_bytes, NULL,
+					 message->body);
+			break;
+		case MQ_MESSAGE_TYPE_DIRECT_RPC:
+			{
+				amqp_bytes_t reply_to = amqp_cstring_bytes(message->reply_to);
+				res = mq_publish(message->exchange, AMQP_EXCHANGE_TYPE_DIRECT,
+					 message->routing_key, headers, num_of_headers,
+					 message->expiration_ms, reply_to,
+					 message->correlation_id, message->body);
+			break;
+			}
+		case MQ_MESSAGE_TYPE_FANOUT:
+			res = mq_publish(message->exchange, AMQP_EXCHANGE_TYPE_FANOUT,
+					 NULL, headers, num_of_headers, message->expiration_ms,
+				  	 amqp_empty_bytes, NULL, message->body);
```

Comment:
Could you to remove these extra white spaces between tabs?
```suggestion
					 amqp_empty_bytes, NULL, message->body);
```